### PR TITLE
Muclight config fix

### PR DESCRIPTION
**Pull request by NelsonVides**
This PR addresses #4173

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/200642) / elasticsearch_and_cassandra_mnesia / 29d818676274df871699ddd89ada5e8322444a1d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4178/200642/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4178/200642/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%406c53cb09ac7f.2023-11-22_16.33.04/index.html)
**OK: 369** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

File: src/muc_light/mod_muc_light_room_config.erl [hunk 1]
```diff
@@ -91,14 +95,14 @@ take_next_kv(RawConfig, [{_KeyBin, Default, Key, _Type} | RSchema]) ->
...
-to_binary_kv(Config, ConfigSchema) ->
-    ConfigWithSchema = lists:zip(lists:sort(Config), lists:keysort(3, ConfigSchema)),
-    [{KeyBin, value2b(Val, Type)} || {{Key, Val}, {KeyBin, _Default, Key, Type}} <- ConfigWithSchema].
-
-%%====================================================================
-%% Internal functions
-%%====================================================================
+take_next_binary_kv([{Key, ValBin} | RRawConfig], [{KeyBin, _Default, Key, Type} | RSchema]) ->
+    try {value, RRawConfig, RSchema, {KeyBin, value2b(ValBin, Type)}}
+    catch _:_ -> {error, {KeyBin, type_error}}
```
Comment (arcusfelis) [inline src/muc_light/mod_muc_light_room_config.erl:100]: I wanted to write "Why so complicated" and I after I see the old take_next_kv doing the same.
{error, _} is not covered by tests even :)

take_next logic is kinda hard to follow, in the perfect world it should be just simple maps:merge(NewConfig, maps:merge(OldConfig, Default)) 
or something.

File: src/muc_light/mod_muc_light_room_config.erl [hunk 2]
```diff
@@ -91,14 +95,14 @@ take_next_kv(RawConfig, [{_KeyBin, Default, Key, _Type} | RSchema]) ->
...
+    end;
+take_next_binary_kv(RawConfig, [{KeyBin, Default, _Key, _Type} | RSchema]) ->
+    {default, RawConfig, RSchema, {KeyBin, Default}};
+take_next_binary_kv([{KeyBin, _} | _], _) ->
```
Comment (chrzaszcz) [inline src/muc_light/mod_muc_light_room_config.erl:104]: Just a minor naming inconsistency
```suggestion
take_next_binary_kv([{Key, _} | _], _) ->
```

**Review by arcusfelis — APPROVED** [review]
looks ok I guess.
